js/generate.js

```diff
@@ -76,10 +77,208 @@ const maturityLevels = {
 };
 
 /**
+ * Lists of columns in generated tables per type of table
+ *
+ * This structure may be completed or overridden in `toc.json` files.
+ */
+const tableColumnsPerType = {
+  'well-deployed': ['feature', 'spec', 'maturity', 'impl'],
+  'in-progress': ['feature', 'spec', 'maturity', 'impl'],
+  'exploratory-work': ['feature', 'spec', 'group', 'implintents'],
+  'versions': ['feature', 'spec', 'maturity', 'versions']
+};
+
+/**
+ * Helper function that expands column definitions into an object structure
+ * (used to allow shortcuts in table columns definitions in tableColumnsPerType
+ * and in custom table definitions that may appear in toc.json.
+ */
+const expandColumns = function (columns, tr) {
+  return (columns || [])
+    .map(column => {
+      if (Object.prototype.toString.call(column) === '[object String]') {
+        return {
+          type: column,
+          title: tr.columns[column]
+        };
+      }
+      else if (!column.type) {
+        console.warn('Skip column definition as `type` property is missing');
+        return null;
+      }
+      else if (!column.title) {
+        column.title = tr.columns[column.type];
+        return column;
+      }
+      else {
+        return column;
+      }
+    })
+    .filter(column => !!column)
+    .map(column => {
+      if (!column.title) {
+        console.warn('No column title found for column type "' + column.type + '" in "' + lang + '"');
+        column.title = column.type;
+      }
+      return column;
+    })
+    .map(column => {
+      if (!tableColumnGenerators[column.type]) {
+        console.warn('Skip unknown column type "' + column.type + '"');
+        return null;
+      }
+      column.generateCell = tableColumnGenerators[column.type];
+      return column;
+    })
+    .filter(column => !!column);
+};
+
+/**
  * Known browsers
  */
 const browsers = ['firefox', 'chrome', 'edge', 'safari', 'webkit'];
 
+/**
+ * Code to call to generate a cell of the given type
+ *
+ * Generators should be called with an object that has the following properties:
```

Comment:
Personally I'd prefer not to use the word "generator" here, because of the potential confusion with [generator function](https://tc39.github.io/ecma262/#sec-generator-function-definitions), but I don't have a better name either. 🤷‍♂️

Reply:
I used "creator" to avoid any confusion. Not that it's much better, but that's all I could find ;)